pkg/porter/generateManifest.go

```diff
@@ -111,15 +113,83 @@ func (p *Porter) generateInternalManifest(ctx context.Context, opts BuildOptions
 			}
 		}
 		return e.SetValue(path+"digest", digest.String())
-
 	})
 	if err != nil {
 		return err
 	}
 
+	if p.IsFeatureEnabled(experimental.FlagDependenciesV2) {
+		if err = p.resolveDependencyDigest(ctx, e, regOpts); err != nil {
+			return err
+		}
+	}
+
 	return e.WriteFile(build.LOCAL_MANIFEST)
 }
 
+func (p *Porter) resolveDependencyDigest(ctx context.Context, e *yaml.Editor, opts cnabtooci.RegistryOptions) error {
+	// find all referenced dependencies that does not have digest specified
+	// get the digest for all of them and update the manifest with the digest
+	return e.WalkNodes(ctx, "dependencies.requires.*", func(ctx context.Context, nc *yqlib.NodeContext) error {
```

Comment:
My only concern is
1) Should we resolve dependencies that aren't required (I feel yes?) - and:
2) Are there any concerns with this and  "shared dependencies", where the dependency is already installed and running (I think this should be okay)

Reply:
1. I'm not sure I fully understand what you mean with "dependencies not required"
2. I think it should be okay too, but I will create a test to verify it. It makes sense long term anyway

Reply:
2. Should be covered by the existing integration for shared dependencies. Do you agree?